lib/Builtins.cpp

```diff
@@ -258,7 +257,8 @@ Builtins::FunctionInfo::getParameter(size_t _arg) const {
 
 // Test for OCL Sampler parameter type
 bool Builtins::ParamTypeInfo::isSampler() const {
-  return type_id == Type::StructTyID && name == "ocl_sampler";
+  return type_id == Type::StructTyID &&
+         (name == "ocl_sampler" || name == "opencl.sampler_t");
```

Comment:
Is this a result of just using the struct name?

Reply:
Now in ReplaceOpenCLBuiltinPass when using Builtins::GetMangledFunctionName to generate replacement functions (such as replacing read_imageh with read_imagef) the actual arguments are used to mangle the name. So subsequent Unmangling will have a different name for sampler struct. This is the actual name of the struct when we see it in the IR.